### PR TITLE
Add support for androidx.paging

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

Add support for androidx.paging

#### WHY

Infinite lists are relevant for message inboxes etc.

#### HOW

Mostly a port of https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:paging/paging-compose/src/main/java/androidx/paging/compose/LazyPagingItems.kt

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

https://user-images.githubusercontent.com/231923/203827492-9c512947-9c1c-4d37-a9fe-90c341522ec1.mp4

### CODE REVIEW

File: sample/src/main/java/com/google/android/horologist/paging/PagingScreen.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                    text = "Waiting for items to load from the backend",
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .wrapContentWidth(Alignment.CenterHorizontally)
+                )
+            }
+        }
+
+        items(lazyPagingItems) { item ->
+            TitleCard(onClick = { /*TODO*/ }, title = {
```
Comment (luizgrp) [inline sample/src/main/java/com/google/android/horologist/paging/PagingScreen.kt:74]: intend to merge the TODO?

File: sample/src/main/java/com/google/android/horologist/paging/PagingScreen.kt [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                )
+            }
+
+            override fun getRefreshKey(state: PagingState<Int, String>): Int? {
+                return (
+                    (
+                        state.anchorPosition
+                            ?: 0
+                        ) - state.config.initialLoadSize / 2
+                    ).coerceAtLeast(0)
```
Comment (luizgrp) [inline sample/src/main/java/com/google/android/horologist/paging/PagingScreen.kt:154]: nit: weird formatting

**Review by luizgrp — APPROVED** [review]
LGTM - screenshot/video in the description would be nice

**Comment by yschimke** [comment]
> LGTM - screenshot/video in the description would be nice

done

**Comment by luizgrp** [comment]
nice! the loading indicator seem static in the vid, are the animations disabled in that device?

**Comment by yschimke** [comment]
> nice! the loading indicator seem static in the vid, are the animations disabled in that device?

Not sure, I noticed that as well.  I'll try on another emulator.

**Comment by yschimke** [comment]
Some issues with placeholders, but that is sample code only.

**Pull request merged**